app/web/features/profile/view/UserOverview.test.tsx

```diff
@@ -0,0 +1,119 @@
+import { render } from "@testing-library/react";
+import { HostingStatus, MeetupStatus } from "proto/api_pb";
+import wrapper from "test/hookWrapper";
+import { addDefaultUser } from "test/utils";
+
+import {
+  COMMUNITY_STANDING,
+  hostingStatusLabels,
+  meetupStatusLabels,
+  VERIFICATION_SCORE,
+} from "../constants";
+import { ProfileUserProvider } from "../hooks/useProfileUser";
+import UserOverview from "./UserOverview";
+
+describe("UserOverview", () => {
+  beforeEach(() => {
+    addDefaultUser();
+  });
+
+  describe("when user is missing", () => {
+    it("doesn't fail", () => {
+      render(<UserOverview />, { wrapper });
+    });
+  });
+
+  describe("when user is loaded and provided via context", () => {
+    it("should display the user name", () => {
+      const { getByText } = render(
+        <ProfileUserProvider user={defaultUser}>
+          <UserOverview showHostAndMeetAvailability={false} />
+        </ProfileUserProvider>,
+        { wrapper }
+      );
+      expect(getByText(defaultUser.name)).toBeInTheDocument();
+    });
+
+    it("should display the user location", () => {
+      const { getByText } = render(
+        <ProfileUserProvider user={defaultUser}>
+          <UserOverview showHostAndMeetAvailability={false} />
+        </ProfileUserProvider>,
+        { wrapper }
+      );
+      expect(getByText(defaultUser.city)).toBeInTheDocument();
+    });
+
+    it("should display the hosting and meeting status only when showHostAndMeetAvailability is true", () => {
+      const expectedLabelHosting =
+        hostingStatusLabels[defaultUser.hostingStatus as HostingStatus];
+      const expectedLabelMeeting =
+        meetupStatusLabels[defaultUser.meetupStatus as MeetupStatus];
+
+      const { queryByText, getByText, rerender } = render(
+        <ProfileUserProvider user={defaultUser}>
+          <UserOverview showHostAndMeetAvailability={false} />
+        </ProfileUserProvider>,
+        { wrapper }
+      );
+      expect(queryByText(expectedLabelHosting)).not.toBeInTheDocument();
+      expect(queryByText(expectedLabelMeeting)).not.toBeInTheDocument();
+      rerender(
+        <ProfileUserProvider user={defaultUser}>
+          <UserOverview showHostAndMeetAvailability />
+        </ProfileUserProvider>
+      );
+      expect(getByText(expectedLabelHosting)).toBeInTheDocument();
+      expect(getByText(expectedLabelMeeting)).toBeInTheDocument();
+      rerender(
+        <ProfileUserProvider user={defaultUser}>
+          <UserOverview />
+        </ProfileUserProvider>
+      );
```

Comment:
I think my comment of defaulting to `false` instead in the component code has another additional benefit, you only need to test the case of being explicitly passed in rather than the missing case too.

Reply:
totally